src/ecs/resources/input/axis.rs

```diff
@@ -0,0 +1,11 @@
+use super::Button;
+
+/// Represents an axis made up of digital inputs, like W and S or A and D.
+/// Two of these could be analogous to a DPAD.
+#[derive(Serialize, Deserialize)]
+pub struct Axis {
+    /// Positive button, when pressed down axis value will return 1 if `neg` is not pressed down.
```

Comment:
@Xaeroxe A Wiimote is a good example.  It gets exposed as a device with (as I recall) 7 buttons, 3 unconstrained axis (translation in 3d space), 3 constained axis (rotation in 3d space), force feedback, and a speaker.

In addition, the Steam Controller.  It can expose itself to a game as an unbounded number of buttons (well I think 256 max), some rotation axis (constrained), and the 2 touchpads on top can become any number of buttons, axis (both constrained or unconstrained) and more, depending on how it is set up.

Among others as well.

For note, the 'hardware' in my system was not even exposed as keyboards/mice/gamepads/whateverCustomThings, but rather just exposed as an Input device (with those 3 exposed things, and a few more things, the Output devices were things like monitors and InputOutput devices were things like my keyboard that had both inputs (boolean and a single constrained axis, yes on a keyboard) and outputs of a texture (each button can be colored on my keyboard mapped to I think it was a 28x96 RGB image that could be sent to it).  Everything was 'tagged' (with my Atom64 type) so like my keyboard was tagged as both a `KEYBOARD` and a `DISPLAY` (of size 28x96) and a couple other things.  My mouse is similar but it has 2 unconstrained axis, 17 buttons, and it was also an Output device because it accepted commands to be able to change the DPI and colors on the fly, yes it was also a display type.

Everything that used the inputs (I've talked about my mapper before) had no clue what the device type was, they just took inputs and translated them to game commands, easily remappable.  The hardware devices were mapped in my 'virtual' filesystem at `/dev/whatever` (potentially in multiple places there depending on its tags), and commands 'to' them or callbacks 'from' them could be routed around the virtual filesystem to issue commands to the engine (everything had a mount somewhere in the virtual filesystem and you could tie things together with impunity), even the ECS system had a mount in the filesystem (even the VFS was an ECS, a game level would be another distinct ECS, perhaps even a singular entity in the level could have its own embedded ECS for very complex entities, I routinely hit 4 levels deep of ECS).  But I've talked about the VFS in depth elsewhere in some other issues around.  ^.^;

There are a *lot* of weird devices out there, my system ended up being able to handle them all (not the prettiest C++ code, but it worked really well and it worked fast).  :-)